src/Idris/Pretty.idr

```diff
@@ -302,7 +302,7 @@ renderWithoutColor doc = do
   consoleWidth <- getConsoleWidth
   opts <- case consoleWidth of
                Nothing => do cols <- coreLift getTermCols
-                             pure $ MkLayoutOptions (AvailablePerLine cols 1)
+                             pure $ MkLayoutOptions (if cols == 0 then Unbounded else AvailablePerLine cols 1)
                Just 0 => pure $ MkLayoutOptions Unbounded
                Just cw => pure $ MkLayoutOptions (AvailablePerLine (cast cw) 1)
```

Comment:
Can you take this opportunity to lift this to the toplevel & then call that
new function in both `do` blocks rather than having duplicated code? Thanks!

Reply:
So do you want a function
```idris
getRealizedPageWidth : {auto o : Ref ROpts REPLOpts} -> Core PageWidth
```
that will be called from both render and renderWithoutColor. Does that have an acceptable name and type signature?

Reply:
Looks good to me!